api/tasks_test.go

```diff
@@ -0,0 +1,185 @@
+			req.Header.Set("Authorization", validServiceAuthToken)
+			resp := httptest.NewRecorder()
+
+			createTaskHandler.ServeHTTP(resp, req)
+
+			Convey("Then an empty search reindex job is returned with status code 404 because the job id was invalid", func() {
+				So(resp.Code, ShouldEqual, http.StatusNotFound)
+				errMsg := strings.TrimSpace(resp.Body.String())
+				So(errMsg, ShouldEqual, "Failed to find job that has the specified id")
+			})
+		})
+	})
+
+	Convey("Given an API that can create valid search reindex tasks and store their details in a Data Store", t, func() {
+		apiInstance := api.Setup(ctx, mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{})
+		createTaskHandler := apiInstance.CreateTaskHandler(ctx)
+
+		Convey("When the tasks endpoint is called to create and store a new reindex task", func() {
+			req := httptest.NewRequest("POST", fmt.Sprintf("http://localhost:25700/jobs/%s/tasks", validJobID1), bytes.NewBufferString(
+				fmt.Sprintf(createTaskPayloadFmt, emptyTaskName)))
+			req.Header.Set("Content-Type", "application/json")
+			req.Header.Set("Authorization", validServiceAuthToken)
+			resp := httptest.NewRecorder()
+
+			createTaskHandler.ServeHTTP(resp, req)
+
+			Convey("Then an empty search reindex job is returned with status code 400 because the task name is empty", func() {
+				So(resp.Code, ShouldEqual, http.StatusBadRequest)
+				errMsg := strings.TrimSpace(resp.Body.String())
+				So(errMsg, ShouldEqual, "invalid request body")
+			})
+		})
+	})
+
+	Convey("Given an API that can create valid search reindex tasks and store their details in a Data Store", t, func() {
+		apiInstance := api.Setup(ctx, mux.NewRouter(), dataStorerMock, &apiMock.AuthHandlerMock{})
+		createTaskHandler := apiInstance.CreateTaskHandler(ctx)
+
+		Convey("When the tasks endpoint is called to create and store a new reindex task", func() {
+			req := httptest.NewRequest("POST", fmt.Sprintf("http://localhost:25700/jobs/%s/tasks", validJobID1), bytes.NewBufferString(
+				fmt.Sprintf(createTaskPayloadFmt, invalidTaskName)))
+			req.Header.Set("Content-Type", "application/json")
+			req.Header.Set("Authorization", validServiceAuthToken)
+			resp := httptest.NewRecorder()
+
+			createTaskHandler.ServeHTTP(resp, req)
+
+			Convey("Then an empty search reindex job is returned with status code 400 because the task name is invalid", func() {
+				So(resp.Code, ShouldEqual, http.StatusBadRequest)
+				errMsg := strings.TrimSpace(resp.Body.String())
+				So(errMsg, ShouldEqual, "invalid request body")
+			})
+		})
+	})
+}
+
+// Create Task Payload
+var createTaskPayloadFmt = `{
+	"task_name": "%s",
+	"number_of_documents": 5
+}`
+
+func ExpectedTask(jobID string, lastUpdated time.Time, numberOfDocuments int, taskName string) (models.Task, error) {
+	cfg, err := config.Get()
+	if err != nil {
+		return models.Task{}, fmt.Errorf("%s: %w", errors.New("unable to retrieve service configuration"), err)
```

Comment:
There's no need to use errors.New() here, as it's being read as it's string interpretation by %s. Simply:
```
 fmt.Errorf("%s: %w", "unable to retrieve service configuration", err)
 ```
 or even
 ```
  fmt.Errorf("unable to retrieve service configuration: %w", err)
  ```
  will produce the same result

Reply:
thanks, that's helpful - done